activerecord/lib/active_record/connection_adapters/abstract/schema_statements.rb

```diff
@@ -1182,6 +1182,58 @@ def remove_check_constraint(table_name, expression = nil, **options)
         execute schema_creation.accept(at)
       end
 
+      # Returns an array of exclusion constraints for the given table.
+      # The exclusion constraints are represented as ExclusionConstraintDefinition objects.
+      def exclusion_constraints(table_name)
+        raise NotImplementedError
+      end
+
+      # Adds a new exclusion constraint to the table. +expression+ is a String
+      # representation of a list of exclusion elements and operators.
+      #
+      #   add_exclusion_constraint :products, "price WITH =, availability_range WITH &&", using: :gist, name: "price_check"
+      #
+      # generates:
+      #
+      #   ALTER TABLE "products" ADD CONSTRAINT price_check EXCLUDE USING gist (price WITH =, availability_range WITH &&)
+      #
+      # The +options+ hash can include the following keys:
+      # [<tt>:name</tt>]
+      #   The constraint name. Defaults to <tt>excl_rails_<identifier></tt>.
+      def add_exclusion_constraint(table_name, expression, **options)
+        return unless supports_exclusion_constraints?
```

Comment:
I am not sure what the usual approach in Rails is here, but thinking out loud - I feel like you'd want your migration to fail noisely, not silently, if you ran a command that your DB doesn't support.

Reply:
that's a great point

Reply:
As with some of your other comments, I don't necessarily disagree with them, but I also think there is a lot of value in consistency of this functionality with the check constraint functionality that already exists in AR, which is why I did this explicit `return`.